src/tabbar.cpp

```diff
@@ -254,3 +279,65 @@ void TabBar::mousePressEvent(QMouseEvent *event)
        QTabBar::mousePressEvent(event);
     }
 }
+
+void TabBar::paintEvent(QPaintEvent *e)
+{
+    // Please keep it in sync with resources/css/style.css
+    // QTabBar::tab:selected { background-color: <value> }
+    const QColor selected_tab_bg_color = QColor(Qt::white);
+
+    // first, let Qt draw QTabBar normally
+    QTabBar::paintEvent(e);
+
+    // Then apply fade-out effect for long tab title on top:
+    QPainter p(this);
+
+    for (int i = 0; i < count(); ++i) {
+        bool need_fade_out = tabData(i).toBool();
+        if (! need_fade_out)
+            continue;
+
+        QStyleOptionTab tab;
+        initStyleOption(&tab, i);
+
+        QRect textRect = style()->subElementRect(QStyle::SE_TabBarTabText, &tab, this);
+
+        QRect tail = textRect;
+        int offset = textRect.width() * 0.8;
+        if (tab.direction == Qt::RightToLeft) {
```

Comment:
It is nice to handle the LeftToRight text but I haven't succeed to test it.
I've tried to run kiwix-desktop in Arabic and use arabic content but the fade-out is always on the right.
(But the title is correctly cut on the left)

Reply:
`tab.direction` was 'widget layout', not `text layout`. Fixed in the current revision (8759c81e).